src/remote/utils.go

```diff
@@ -0,0 +1,88 @@
+package remote
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"time"
+
+	pb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/bazelbuild/remote-apis/build/bazel/semver"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// digestMessage calculates the digest of a proto message as described in the
+// Digest message's comments.
+func digestMessage(msg proto.Message) *pb.Digest {
+	digest, _ := digestMessageContents(msg)
+	return digest
+}
+
+// digestMessageContents is like DigestMessage but returns the serialised contents as well.
+func digestMessageContents(msg proto.Message) (*pb.Digest, []byte) {
+	b, err := proto.Marshal(msg)
+	if err != nil {
+		// Not really sure if there is a valid possibility to bring us here (given that
+		// the messages in question have no required fields) so assume it won't happen :)
+		log.Fatalf("Failed to marshal message: %s", err)
```

Comment:
I'm not a fan of `logger.Fatal` - os.Exit is a bit of a blunt instrument. Could we panic() instead?

Reply:
I'm not really a fan of spitting out a big stacktrace at the user if they can't do anything with it. As the comment mentions though, I'm not sure what can even cause it to get here...